diff --git a/configure.ac b/configure.ac
index 50ef80e..3c344e9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -387,8 +387,13 @@ if test x$ac_cv_c_compiler_gnu = xyes ; then
 		CXXFLAGS="$CXXFLAGS -fvisibility=hidden"
 		fi
 
+	# Bugfix for OSX 10.9 Clang. Thanks to Serguei E. Leontiev. http://sourceforge.net/p/flac/bugs/405/
 	if test "x$GCC_MAJOR_VERSION$GCC_MINOR_VERSION" = "x42" ; then
-		XIPH_ADD_CFLAGS([-fgnu89-inline])
+		if $CC --version 2>&1| grep -q clang ; then
+		    true
+		else
+		    XIPH_ADD_CFLAGS([-fgnu89-inline])
+		    fi
 		fi
 	fi
 
